extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -248,9 +247,11 @@ public static void startServerAfterFailedStart() {
         Supplier<Vertx> supplier = VertxCoreRecorder.getVertx();
         Vertx vertx;
         if (supplier == null) {
-            VertxConfiguration vertxConfiguration = new VertxConfiguration();
-            ConfigInstantiator.handleObject(vertxConfiguration);
-            vertx = VertxCoreRecorder.recoverFailedStart(vertxConfiguration).get();
+            // TODO: we can't do this anymore
+            //VertxConfiguration vertxConfiguration = new VertxConfiguration();
+            //ConfigInstantiator.handleObject(vertxConfiguration);
```

Comment:
I don't really know how I can do this. AFAICS, it parses specifically the Vert.x HTTP config to be able to start a Vert.x HTTP server without triggering other errors if possible.
This is a big hack, I'm unsure we want to keep this behavior? If so, I think it's important for @radcortez to have a look anyway.

Reply:
Yes, the idea was that we would still try to get a minimal dev mode up even if something is very wrong. It is useful for sure and we need to keep this feature. I commented it out so that I was able to compile the extension and get the tests running.

Reply:
Well, this still relies on `SmallRyeConfig` to populate the object, so if you have an invalid configuration it would still fail:

```
Caused by: java.lang.RuntimeException: java.lang.IllegalArgumentException: SRCFG00039: The config property quarkus.http.port with the config value "x" threw an Exception whilst being converted SRCFG00029: Expected an integer value, got "x"
	at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:139)
	at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:78)
	at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:57)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServerAfterFailedStart(VertxHttpRecorder.java:264)
	... 9 more
```

Unless I'm missing something doing:

```java
SmallRyeConfig config = ConfigProvider.getConfig().unwrap(SmallRyeConfig.class);
VertxConfiguration vertxConfiguration = config.getConfigMapping(VertxConfiguration.class);
```

Should be enough.

Reply:
This is now supported automatically with https://github.com/smallrye/smallrye-config/pull/939